src/app/item-review/d1-dtr-api-types.ts

```diff
@@ -0,0 +1,117 @@
+import { DtrReviewer, DimWorkingUserReview, DtrRating, DimUserReview } from "./dtr-api-types";
+
+/**
+ * A fetch request for a single weapon. Expected to be part of a bulk (array) fetch request.
+ */
+export interface D1ItemFetchRequest {
+  /** Reference ID for the weapon. */
+  referenceId: string;
+  /**
+   * The roll (available perks).
+   * Note that we only send random perks, so exotics, some raid weapons and other weapons don't pass this.
+   */
+  roll: string | null;
+}
+
+/** Bulk fetch response for a single weapon. */
+export interface D1ItemFetchResponse {
+  /** Reference ID for the weapon. */
+  referenceId: string;
+  /**
+   * The roll (available perks).
+   * Note that we only send random perks, so exotics, some raid weapons and other weapons don't pass this.
+   */
+  roll: string | null;
+  /** The rating from DTR. We use this. */
+  rating?: number;
+  /** The number of ratings that DTR has for the weapon (roll). */
+  ratingCount: number;
+  /** The number of highlighted ratings that DTR has for the weapon (roll). */
+  highlightedRatingCount: number;
+}
+
+/** A request for reviews for a single weapon. */
+export interface D1ItemReviewRequest {
+  /** Reference ID for the weapon. */
+  referenceId: string;
+  /**
+   * The roll (available perks).
+   * Note that we only send random perks, so exotics, some raid weapons and other weapons don't pass this.
+   */
+  roll: string | null;
+  /** What perks does the user have selected on it? */
+  selectedPerks: string | null;
+  /** What's the particular instance ID of the weapon? */
+  instanceId: string;
+}
+
+/** The local user's working rating. */
+export interface WorkingD1Rating extends DimWorkingUserReview {
+  /**
+   * What's the weapon's rating?
+   * In D1, ratings are on a scale of 1-5.
+   */
+  rating: number;
+  /** Pros - set aside for future use. */
+  pros: string;
+  /** Cons - set aside for future use. */
+  cons: string;
+  /** Text for the review.
+   * Optional.
+   */
+  review: string;
+}
+
+/** A single user's review for a D1 weapon. */
+export interface D1ItemUserReview extends DimUserReview {
```

Comment:
Elsewhere I've been putting the D1 and D2 specific subclasses right next to the general type, so it's easy to see what differs between them.

Reply:
Maybe I'm confused, but this seems incompatible with a suggestion above (to put request/response types down in the fetch class)?

Reply:
I don't think it's inconsistent. The request/response classes shouldn't really be related since they're different APIs, while the massaged user review structure inherits from the same base properties.